aries-site/src/examples/templates/filtering/FilteringWithDropButton.js

```diff
@@ -0,0 +1,434 @@
+import React, { useContext, useRef, useState } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import {
+  Anchor,
+  Box,
+  Button,
+  Card,
+  CheckBoxGroup,
+  DropButton,
+  FormField,
+  Form,
+  Header,
+  Heading,
+  Grid,
+  ResponsiveContext,
+  Text,
+  TextInput,
+} from 'grommet';
+import { Filter, Search } from 'grommet-icons';
+
+const allData = [
+  {
+    status: 'Online',
+    name: 'd6db7379-c71c-4e88-ab26-ab788cdccaa5',
+    ram: '16',
+  },
+  {
+    status: 'Offline',
+    name: 'bbde1017-2de5-4a6a-9fa0-7802dafc852c',
+    ram: '32',
+  },
+  {
+    status: 'Online',
+    name: '931cdda8-d5f2-4294-85b5-da8ff8100fa5',
+    ram: '128',
+  },
+  {
+    status: 'Online',
+    name: '09ff2e6f-b452-4943-b6b6-a13d70a20117',
+    ram: '128',
+  },
+  {
+    status: 'Offline',
+    name: '8a617640-69ca-4d6f-8503-4b272426728f',
+    ram: '128',
+  },
+  {
+    status: 'Offline',
+    name: '16c9aae6-6732-4f95-b139-3363701e7770',
+    ram: '32',
+  },
+];
+
+const StyledCard = styled(Card)`
+  transition: all 0.3s ease-in-out;
+  :focus,
+  :hover {
+    transform: scale(1.01, 1.01);
+  }
+`;
+
+const StyledTextInput = styled(TextInput).attrs(() => ({
+  'aria-labelledby': 'search-icon',
+}))``;
+
+const StyledButton = styled(Button)`
+  border: 1px solid
+    ${({ theme }) => theme.global.colors.border[theme.dark ? 'dark' : 'light']};
+  &:hover {
+    background: transparent;
+  }
+`;
+
+const defaultFilters = {};
+const ramOptions = ['16', '32', '128'];
+const statuses = ['Online', 'Offline'];
+
+export const FilteringWithDropButton = () => {
+  const [data, setData] = useState(allData);
+  const [filtering, setFiltering] = useState(false);
+  const [searchFocused, setSearchFocused] = useState(false);
+  const [filters, setFilters] = useState(defaultFilters);
+  const [search, setSearch] = useState();
+  const size = useContext(ResponsiveContext);
+  const inputRef = useRef();
+
+  const filterData = (array, criteria, searchValue = search) => {
+    if (Object.keys(criteria).length) setFiltering(true);
+    else setFiltering(false);
+    setFilters(criteria);
+
+    let filterResults;
+    const filterKeys = Object.keys(criteria);
+    filterResults = array.filter(item => {
+      // validates all filter criteria
+      return filterKeys.every(key => {
+        // ignores non-function predicates
+        if (typeof criteria[key] !== 'function') return true;
+        return criteria[key](item[key]);
+      });
+    });
+
+    if (searchValue) {
+      filterResults = filterResults.filter(o =>
+        Object.keys(o).some(
+          k =>
+            typeof o[k] === 'string' &&
+            o[k].toLowerCase().includes(searchValue.toLowerCase()),
+        ),
+      );
+    }
+    return filterResults;
+  };
+
+  return (
+    <Box
+      fill
+      gap="medium"
+      margin="auto"
+      overflow="auto"
```

Comment:
Move to `<Results />`?

Reply:
Without overflow="auto", the example will stretch outside of the Example bounds.

Reply:
Fine to leave here, more was thinking the Results could use an overflow treatment - related: https://github.com/hpe-design/design-system/pull/1084#discussion_r486546883